Sources/ETCD/ETCDClient.swift

```diff
@@ -120,5 +123,129 @@ public final class EtcdClient: @unchecked Sendable {
     public func put(_ key: String, value: String) async throws {
         try await put(key.utf8, value: value.utf8)
     }
+    
+    /// Watches a specified key in the ETCD server.
+    ///
+    /// - Parameters:
+    ///   - key: The key for which the value is watched. Parameter is of type Sequence<UInt8>.
+    ///   - operation: The operation to be run on the WatchAsyncSequence for the key.
+    public func watch<Result>(_ key: some Sequence<UInt8>, _ operation: (WatchAsyncSequence) async throws -> Result) async throws -> Result {
+        let request = [Etcdserverpb_WatchRequest.with { $0.createRequest.key = Data(key) }]
+        let watchAsyncSequence = WatchAsyncSequence(grpcAsyncSequence: watchClient.watch(request))
+        return try await operation(watchAsyncSequence)
+    }
+    
+    /// Watches a specified key in the ETCD server.
+    ///
+    /// - Parameters:
+    ///   - key: The key for which the value is watched. Parameter is of type String.
+    ///   - operation: The operation to be run on the WatchAsyncSequence for the key.
+    public func watch<Result>(_ key: String, _ operation: (WatchAsyncSequence) async throws -> Result) async throws -> Result {
+        try await self.watch(key.utf8, operation)
+    }
+}
+
+public enum EventType {
+    case put
+    case delete
+
+    init(protoType: Int) {
+        switch protoType {
+        case 0:
+            self = .put
+        case 1:
+            self = .delete
+        default:
+            fatalError("Unsupported event type")
+        }
+    }
+
+    var protoType: Int {
+        switch self {
+        case .put:
+            return 0
+        case .delete:
+            return 1
+        }
+    }
+}
+
+public struct KeyValue {
+    public var key: Data
+    public var createRevision: Int64
+    public var modRevision: Int64
+    public var version: Int64
+    public var value: Data
+    public var lease: Int64
+
+    init(protoKeyValue: Etcdserverpb_KeyValue) {
+        self.key = protoKeyValue.key
+        self.createRevision = protoKeyValue.createRevision
+        self.modRevision = protoKeyValue.modRevision
+        self.version = protoKeyValue.version
+        self.value = protoKeyValue.value
+        self.lease = protoKeyValue.lease
+    }
+}
 
+
+public struct WatchEvent {
+    public var kv: KeyValue
+    public var prevKV: KeyValue?
+
+    init(protoEvent: Etcdserverpb_Event) {
+        self.kv = KeyValue(protoKeyValue: protoEvent.kv)
+        if let protoPrevKV = protoEvent.hasPrevKv ? protoEvent.prevKv : nil {
+            self.prevKV = KeyValue(protoKeyValue: protoPrevKV)
+        } else {
+            self.prevKV = nil
+        }
+    }
+}
+
+
+public struct WatchAsyncSequence: AsyncSequence {
```

Comment:
Can we please move this to a separate file and add some docs to it

Reply:
Added!